src/contexts/bridge.context.tsx

```diff
@@ -568,14 +588,110 @@ const BridgeProvider: FC<PropsWithChildren> = (props) => {
     [getBridges, refreshBridges]
   );
 
+  const cleanPendingTxs = useCallback(
+    (bridges: Bridge[]): Promise<void> => {
+      if (!env) {
+        return Promise.resolve();
+      }
+
+      const pendingTxs = storage.getPendingTxs(env);
+      const isPendingDepositInApiBridges = (depositTxHash: string) => {
+        return bridges.find((bridge) => {
+          return (
+            (bridge.status === "initiated" || bridge.status === "on-hold") &&
+            bridge.depositTxHash === depositTxHash
+          );
+        });
+      };
+      const isPendingClaimInApiBridges = (claimTxHash: string) => {
+        return bridges.find((bridge) => {
+          return bridge.status === "completed" && bridge.claimTxHash === claimTxHash;
+        });
+      };
+
+      return Promise.all(
+        pendingTxs.map(async (pendingTx) => {
+          if (
+            pendingTx.status === "pending-deposit" &&
+            isPendingDepositInApiBridges(pendingTx.depositTxHash)
+          ) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+          if (
+            pendingTx.status === "pending-claim" &&
+            isPendingClaimInApiBridges(pendingTx.claimTxHash)
+          ) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+
+          const txHash =
+            pendingTx.status === "pending-deposit"
+              ? pendingTx.depositTxHash
+              : pendingTx.claimTxHash;
+          const provider =
+            pendingTx.status === "pending-deposit"
+              ? pendingTx.from.provider
+              : pendingTx.to.provider;
+          const tx = await provider.getTransaction(txHash);
+
+          if (isTxCanceled(tx)) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+
+          if (isTxMined(tx)) {
+            const txReceipt = await provider.getTransactionReceipt(txHash);
+
+            if (hasTxBeenReverted(txReceipt)) {
+              return storage.removePendingTx(pendingTx.depositTxHash);
+            }
+          }
+
+          if (Date.now() > pendingTx.timestamp + PENDING_TX_TIMEOUT) {
+            return storage.removePendingTx(pendingTx.depositTxHash);
+          }
+        })
+      ).then();
```

Comment:
:neutral_face:

Reply:
Ah, I see! I said about the TS shut up because I thought it was there to force the function to return `void` instead of `void[]`!

It's perfect now!